lib/mongo/collection/view/readable.rb

```diff
@@ -45,6 +45,14 @@ module Readable
         #
         # @since 2.0.0
         def aggregate(pipeline, options = {})
+          # Because the $merge and $out pipeline stages write documents to the
+          # collection, it is necessary to clear the cache when they are performed.
+          write = pipeline.any? do |op| 
+            op.key?('$out') || op.key?(:$out) ||
+              op.key?('$merge') || op.key?(:$merge)
+          end
```

Comment:
Aggregation has a `write?` method that does this, can it be used to decide whether to clear the cache?

Reply:
Good point!